signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/azure/AzureHttpClientFactory.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing.secp256k1.azure;
+
+import tech.pegasys.web3signer.keystorage.azure.AzureHttpClient;
+import tech.pegasys.web3signer.keystorage.azure.AzureHttpClientParameters;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.time.Duration;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.jetbrains.annotations.VisibleForTesting;
+
+public class AzureHttpClientFactory {
+  private final int CONNECTION_POOL_SIZE_LIMIT = 10;
```

Comment:
would prefer something like CLIENT_CACHE_SIZE as don't think this is a connection pool in the traditional sense.

Also, maybe we want to make this configurable, but doesn't need to be in this PR.